api_tests/preprints/filters/test_filters.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+from nose.tools import *  # flake8: noqa
+import pytest
+from mock import MagicMock
+
+class PreprintsListFilteringMixin(object):
+
+    def setUp(self):
+        super(PreprintsListFilteringMixin, self).setUp()
+
+        self._setUp()
+
+        self.provider_url = '{}filter[provider]='.format(self.url)
+        self.id_url = '{}filter[id]='.format(self.url)
+        self.date_created_url = '{}filter[date_created]='.format(self.url)
+        self.date_modified_url = '{}filter[date_modified]='.format(self.url)
+        self.date_published_url = '{}filter[date_published]='.format(self.url)
+        self.is_published_url = '{}filter[is_published]='.format(self.url)
+
+    def _setUp(self):
+        raise NotImplementedError
+
+    def test_provider_filter_null(self):
+        expected = []
+        res = self.app.get('{}null'.format(self.provider_url), auth=self.user.auth)
+        actual = [preprint['id'] for preprint in res.json['data']]
+        assert_equal(expected, actual)
+
+    def test_provider_filter_equals_returns_one(self):
+        expected = [self.preprint_two._id]
+        res = self.app.get('{}{}'.format(self.provider_url, self.provider_two._id), auth=self.user.auth)
+        actual = [preprint['id'] for preprint in res.json['data']]
+        assert_equal(expected, actual)
+
+    def test_provider_filter_equals_returns_multiple(self):
+        expected = set([self.preprint._id, self.preprint_three._id])
+        res = self.app.get('{}{}'.format(self.provider_url, self.provider._id), auth=self.user.auth)
```

Comment:
I'm not sure I understand this test. Why do you expect to receive multiple results if you are only filtering on one _id?